samza-core/src/main/java/org/apache/samza/execution/JobPlanner.java

```diff
@@ -70,35 +70,20 @@ ExecutionPlan getExecutionPlan() {
 
   /* package private */
   ExecutionPlan getExecutionPlan(String runId) {
+    Map<String, String> generatedConfig = getGeneratedConfig(runId);
 
-    // update application configs
-    Map<String, String> cfg = new HashMap<>();
-    if (StringUtils.isNoneEmpty(runId)) {
-      cfg.put(ApplicationConfig.APP_RUN_ID, runId);
+    // merge user-provided configuration with generated configuration. generated configuration has lower priority.
+    // TODO: This should all be consolidated with ExecutionPlanner after fixing SAMZA-1811
+    Map<String, String> allowedUserConfig = new HashMap<>(userConfig);
+    if (!LegacyTaskApplication.class.isAssignableFrom(appDesc.getAppClass())) {
+      allowedUserConfig.remove(TaskConfig.INPUT_STREAMS()); // must be set using descriptors or operators
```

Comment:
The previous code that did this had a log about the override. Would it be good to put that here too?

Reply:
Sure, will add back logging.